### PR TITLE
feat:  remove unnecessary page number in useUrlFilters, add tenant to resource kind

### PR DESCRIPTION
**Pull request by eshankvaish**
# Description

- add tenant to resource kind
- remove unnecessary page number in useUrlFilters
- add support for confirmation text in Delete Dialog
- add stop propagation for VisibleModal, VisibleModal2 & Modal
- add tenant to resource kind

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] Does this PR require documentation updates?
- [ ] I've updated documentation as required by this PR.
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Common/EmptyState/GenericFilterEmptyState.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { GenericEmptyStateType } from '../Types'
+import noResult from '../../Assets/Img/empty-noresult@2x.png'
+import GenericEmptyState from './GenericEmptyState'
+
+const GenericFilterEmptyState = (
```
Comment (eshankvaish) [inline src/Common/EmptyState/GenericFilterEmptyState.tsx:5]: Would be removed once main is back merged

File: src/Common/Dialogs/DeleteDialog.tsx [hunk 2]
```diff
@@ -13,6 +30,18 @@ export const DeleteDialog: React.FC<DeleteDialogProps> & { Description?: React.F
         props.delete()
     }
 
+    const handleKeyDown = async (event: KeyboardEvent) => {
+        if (event.key === 'Enter' && !isDeleteDisabled) {
+            event.preventDefault()
+
+            if (props.shouldStopPropagation) {
```
Comment (AbhishekA1509) [inline src/Common/Dialogs/DeleteDialog.tsx:37]: Should we move this outside parent if or is it required for something else?

**Review by AbhishekA1509 — CHANGES_REQUESTED** [review]

File: src/Common/Dialogs/DeleteDialog.tsx [hunk 3]
```diff
@@ -21,13 +50,34 @@ export const DeleteDialog: React.FC<DeleteDialogProps> & { Description?: React.F
         props.closeDelete()
     }
 
+    const getLabel = () => (
+        <span>
+            Type
+            <span className="fw-6">&nbsp;&apos;{deleteConfirmationText}&apos;&nbsp;</span>
```
Comment (AbhishekA1509) [inline src/Common/Dialogs/DeleteDialog.tsx:56]: Can add word break here

File: src/Common/Dialogs/DeleteDialog.tsx (hunk 2 above)
Comment (eshankvaish) [reply src/Common/Dialogs/DeleteDialog.tsx:37]: Required only on enter

File: src/Common/Dialogs/DeleteDialog.tsx (hunk 3 above)
Comment (eshankvaish) [reply src/Common/Dialogs/DeleteDialog.tsx:56]: added in https://github.com/devtron-labs/devtron-fe-common-lib/pull/171/commits/3718cfa1471e7e12d902eb5c46a7e867d0e8bd48

**Pull request merged**